### ISSUE TITLE
[member-ordering] report alphabetical sorting for all groups instead of just the first failing group

### ISSUE THREAD

**Issue by flohw** [typescript-eslint/typescript-eslint#8100]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

[link](https://typescript-eslint.io/play/#ts=5.3.2&fileType=.ts&code=MYGwhgzhAEBiD29oG8BQ1oDcwgK4FMAhaAXmgHJs99pDyBudLHAgQVIqoOlYdSYAOAJ3gAXfMHEATaBFFhRAS2C0OlFvj4ZhYidNnylK9mXXU%2BTCLgBGAW0WiAFAEoU0AL5MA5vlEARBTAXN08MYCF8BXxg5A8mCIhfGM9PIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANrhsOPtCGQANHPlZ52SKIBmAQwT4pmrdqGjoUyPvjEAFvt75KZW-EKr1ZqJx58AFRIUKVkfLVZWMABJJlEADzBkSnEmfXxYaERvcMhk1PTMxC9w%2BUg3eHhafLSMrJLSiEiwADFKRHhhZByfSGIqADd0xGr8dNdaXXbOhsbIAGJ%2ByiH8EeQxlzJJ6eFZ0r6lVbJV4VHxramO3bVGnGJYbkYt9fPtq73cpZWR0TIhYdOlxmN1uBwERxOtF%2B-0hQOuPTMfQeTyhiD%2B0ABb2BCK0B2Ww1ozBeTDIIzhH16i0GBKJYxJZJ2FMR-XBaMhtP09Kx8NuUHujwmHK55JBcxZEMQpyc6wxx25TNx-JR0vwsroIpxZS%2BBI1vIW2tW8tF%2B3FbMlRs12iVE11oJemwteqFpMdoJVatdc2hGNhjONuVtjWaAEEyKTkMghN09Qa1hsJvowyhI1Z-ZTY2cHYnwymFVrDmbTvaE0mI8o08zkRNi1ts8ny5a7tTDd7MXWy6nG3yC8dza3Ie3cxXFVWtv3zYOGzHmyNnSNJ529VT8Ya57QF3mrT32UxiS6N8OtaPCbu6fvS0Ou2CJVLuDLE3QD1frbW76qH%2BuL1PQRmn0vf1%2Bi4-tuE6AZudzHn%2BdrxrWYGHtoa5QXM7ofkh%2BzjlKcFXmh4TNAA4og%2BCrNA0a3M0vIUfmK5xq84iEeBYDkZRlH6jOmYTHRxjwU0UTMSxpq9kWMG0JxDFMXxP7HjWIn0dxjGsF24kSSabEYTJXGKbxyliiBpxqaJclKdplYCmOaIwuaBmaZpxk6dRJ57iMVm8kZtnaMu3wOWeTmyTZbm9AJO6OepYlaf5R6mV5nIus5ZEKS5YXhU2rKCeub4erFQaJUlSKRShcqZaUkR%2BUlW72YVuHZeFrHlb5CWlfmKWQhVPiuf5uUoi1ZjFfVDU4NJXURFV7VroN8htW5NjpR%2BY3YD1cV9Tg%2Bl1XFJUNZAs08WAADKhHEaRWWLd29nScgK2HYtNWead51FcNk2BeaN0ab1fUddWwlnS9C2XRmalfaFR3XoWqLopiAOGfdtnvWZYOQhD1mvetGZrgjSOlVdNKntFay3ZVQOPacqN461UPGTDUVcmjq3ozlhNpfecrUxdb0vgz75MyT3XxT9rNsczd1Ax5BIC-jv26dUXNDQTUlS%2BNPMs%2Btz2A5dxPfYrGP5XQouk2tGP-XLc0K4Ll069zYAALKEfYAhdM%2BUnCZw%2BA2zywFNU9jvW7bDGY4a0lOy7PsZv7Xuu2Kx5qQH3tycDqWR6HQeqeZPrmlHYcmsea5p4n7tE9jXLZzHKP5y6hdXsLq4lyMZcxpB01yjXbs3uzHqN%2BHkVtypue0J3nxsb36b9wnMch870dXlnw-YfXdAD4i8dj%2BnuRz9L20wWAzCUC4tiUAAXuMAhMFe0mb9vjD7y4h%2BAwAwofMqwMcUbPmxfynnAj9ATp3ev-fH9B8eP9VQP3wN%2BOYgD34gKsJaAAujiAAvj0BB8hYEQAQXAoAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Repro Code

```TypeScript
class Foo {
  valueB = 'value B';
  valueA = 'value A';

  protected static B = 'value';
  protected static A = 'value';

  submit() { }
  getData() { }
  create() { }
  reset() {}
}
```

### ESLint Config
<details>
<summary>Config</summary>

```javascript
{
  "rules": {
    "@typescript-eslint/member-ordering": [
      "error",
      {
        "default": {
          "order": "alphabetically",
          "memberTypes": [
            // Index signature
            "signature",
            "call-signature",
            // Fields
            "private-static-field",
            "#private-static-field",
            "protected-static-field",
            "public-static-field",
            "private-decorated-field",
            "protected-decorated-field",
            "public-decorated-field",
            "private-instance-field",
            "#private-instance-field",
            "protected-instance-field",
            "public-instance-field",
            "protected-abstract-field",
            "public-abstract-field",
            "private-field",
            "#private-field",
            "protected-field",
            "public-field",
            "static-field",
            "instance-field",
            "abstract-field",
            "decorated-field",
            "field",
            // Accessors
            "private-static-accessor",
            "#private-static-accessor",
            "protected-static-accessor",
            "public-static-accessor",
            "private-decorated-accessor",
            "protected-decorated-accessor",
            "public-decorated-accessor",
            "private-instance-accessor",
            "#private-instance-accessor",
            "protected-instance-accessor",
            "public-instance-accessor",
            "protected-abstract-accessor",
            "public-abstract-accessor",
            "private-accessor",
            "#private-accessor",
            "protected-accessor",
            "public-accessor",
            "static-accessor",
            "instance-accessor",
            "abstract-accessor",
            "decorated-accessor",
            "accessor",

            // Methods
            "public-static-method",
            "protected-static-method",
            "#private-static-method",
            "private-static-method",
            "public-decorated-method",
            "protected-decorated-method",
            "private-decorated-method",
            "public-instance-method",
            "protected-instance-method",
            "private-instance-method",
            "#private-instance-method",
            "public-abstract-method",
            "protected-abstract-method",
            "public-method",
            "protected-method",
            "private-method",
            "#private-method",
            "static-method",
            "instance-method",
            "abstract-method",
            "decorated-method",
            "method",
            // Static initialization
            "static-initialization",
            // Constructors
            "private-constructor",
            "protected-constructor",
            "public-constructor",
            "constructor"
          ]
        }
      }
    ]
  }
}
```

</details>

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Get the complete list of all the errors at once.

If order of member types is not respected and alphabetical order is not respected within the member type we first get the error of the member type order then the error about the alphabetical order. Would be nice if we can have both errors at the same time.

### Actual Result

We have the first error on the property order.

Then start the alphabetical order errors on the protected static properties. When fixed we then have the error on the public properties. Plus we have one alphabetical order error at a time.

Same applies on the methods. When fixing the error on getData before submit we then have create must be before submit the create must be before getData then reset must be before submit.

### Additional Info

I hope it will be an easy enhancement. :slightly_smiling_face: 

Let me know if I need to provide more info.

Additionally, the error reported about where to place the node is not always the more logic to me. But it will require another issue and tied to the algorithm used to detect the errors which may be hard to change and we can deal with it. Let's improve one step at a time. :slightly_smiling_face:

**Comment by bradzacher** [comment]
Note that the rule doesn't exit on the first error [playground](https://typescript-eslint.io/play/#ts=5.3.2&fileType=.ts&code=MYGwhgzhAEBiD29oG8BQ1oDcwgK4FMBhaAXmgCJyBuVdaABwCd4AXfYNgE2ghbBYCWwaACFS0AOTY8%2BCTQxNW7Ljz6DhAQXFScBObQzSCYsjpmj9h3fi2mj%2BaBv10mA7G1X8h0YmUo0AX1QgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANrhsOPtCGQANHPlZ52SKIBmAQwT4pmrdqGjoUyPvjEAFvt75KZW-EKr1ZqJx58AFRIUKVkfLVZWMABJJlEADzBkSnEmfXxYaERvcMhk1PTMxC9w%2BUg3eHhafLSMrJLSiEiwADFKRHhhZByfSGIqADd0xGr8dNdaXXbOhsbIAGJ%2ByiH8EeQxlzJJ6eFZ0r6lVbJV4VHxramO3bVGnGJYbkYt9fPtq73cpZWR0TIhYdOlxmN1uBwERxOtF%2B-0hQOuPTMfQeTyhiD%2B0ABb2BCK0B2Ww1ozBeTDIIzhH16i0GBKJYxJZJ2FMR-XBaMhtP09Kx8NuUHujwmHK55JBcxZEMQpyc6wxx25TNx-JR0vwsroIpxZS%2BBI1vIW2tW8tF%2B3FbMlRs12iVE11oJemwteqFpMdoJVatdc2hGNhjONuVtjWaAEEyKTkMghN09Qa1hsJvowyhI1Z-ZTY2cHYnwymFVrDmbTvaE0mI8o08zkRNi1ts8ny5a7tTDd7MXWy6nG3yC8dza3Ie3cxXFVWtv3zYOGzHmyNnSNJ529VT8Ya57QF3mrT32UxiS6N8OtaPCbu6fvS0Ou2CJVLuDLE3QD1frbW76qH%2BuL1PQRmn0vf1%2Bi4-tuE6AZudzHn%2BdrxrWYGHtoa5QXM7ofkh%2BzjlKcFXmh4TNAA4og%2BCrNA0a3M0vIUfmK5xq84iEeBYDkZRlH6jOmYTHRxjwU0UTMSxpq9kWMG0JxDFMXxP7HjWIn0dxjGsF24kSSabEYTJXGKbxyliiBpxqaJclKdplYCmOaIwuaBmaZpxk6dRJ57iMVm8kZtnaMu3wOWeTmyTZbm9AJO6OepYlaf5R6mV5nIus5ZEKS5YXhU2rKCeub4erFQaJUlSKRShcqZaUkR%2BUlW72YVuHZeFrHlb5CWlfmKWQhVPiuf5uUoi1ZjFfVDU4NJXURFV7VroN8htW5NjpR%2BY3YD1cV9Tg%2Bl1XFJUNZAs08WAADKhHEaRWWLd29nScgK2HYtNWead51FcNk2BeaN0ab1fUddWwlnS9C2XRmalfaFR3XoWqLopiAOGfdtnvWZYOQhD1mvetGZrgjSOlVdNKntFay3ZVQOPacqN461UPGTDUVcmjq3ozlhNpfecrUxdb0vgz75MyT3XxT9rNsczd1Ax5BIC-jv26dUXNDQTUlS%2BNPMs%2Btz2A5dxPfYrGP5XQouk2tGP-XLc0K4Ll069zYAALKEfYAhdM%2BUnCZw%2BA2zywFNU9jvW7bDGY4a0lOy7PsZv7Xuu2Kx5qQH3tycDqWR6HQeqeZPrmlHYcmsea5p4n7tE9jXLZzHKP5y6hdXsLq4lyMZcxpB01yjXbs3uzHqN%2BHkVtypue0J3nxsb36b9wnMch870dXlnw-YfXdAD4i8dj%2BnuRz9L20wWAzCUC4tiUAAXuMAhMFe0mb9vjD7y4h%2BAwAwofMqwMcUbPmxfynnAj9ATp3ev-fH9B8eP9VQP3wN%2BOYgD34gKsJaAAujiAAvj0BB8hYEQAQXAoAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) - it's specifically the alphabetical sorting that short-circuits.

I don't know if there was a reason for this, exactly - the alphabetical sorting was added 4 years ago. I don't think there's any specific reason that it was written that way - I don't see why we couldn't check each group and report on it separately.

I can understand short-circuiting a group after the first member in the grouo is found out-of-order - but not how it works now.

I'm open to someone changing this.

**Comment by flohw** [comment]
Indeed, the member type errors does not exits for them but it exits before checking the alphabetical order.

I don't know if it would be possible to report both at the same time.

I updated the playground lately and multiple times to show both errors. I think I have reported everything I wanted in the latest edit I made.

Thanks for the quick review of the issue.

Do you think the issue could be easy to fix? And located only in the member-ordering module ? If so I can try to fix it but I am far from an expert to the AST and how to handle eslint properly. I just managed to understand a bit better how to configure it ! :sweat_smile:

**Comment by bradzacher** [comment]
I think it's an easy fix - I think this particular block is the problem

https://github.com/typescript-eslint/typescript-eslint/blob/fca72762772833d1d696d9860c535c294ea24245/packages/eslint-plugin/src/rules/member-ordering.ts#L954-L959

Based on a quick browse of the code - I think the problem is that it's using `.some` - which exits as soon as one alpha sort error is found.
